syntax_and_semantics/structs.md

```diff
@@ -11,9 +11,30 @@ struct Point
 end
 ```
 
-## Passing
+Structs inherit from [Value](https://crystal-lang.org/api/Value.html) so they are allocated on the stack and passed by value: when you pass it to methods, return it from methods or assign it to variables, a copy of the value is actually passed (while classes inherit from [Reference](https://crystal-lang.org/api/Reference.html), are allocated on the heap and passed by reference). Therefore you should prefer using structs for immutable data types and/or stateless wrappers of other types, usually for performance reasons to avoid lots of small memory allocations when passing small copies might be more efficient (for more details, see the [performance guide](https://crystal-lang.org/docs/guides/performance.html#use-structs-when-possible)).
 
-A struct is [passed by value](http://crystal-lang.org/api/Value.html) while a class is passed by reference.
+Mutable structs are still allowed, but you should be careful when writing code involving mutability if you want to avoid surprises that are described below.
+
+## Passing by value
+
+A struct is _always_ passed by value, even when you return `self` from the method of that struct:
+
+```crystal
+struct Point
+  def move_right
+    @x += 1
+    self
+  end
+end
+
+p = Point.new(0, 0)
+puts p.move_right.move_right #=> Point(@x=2, @y=0)
+puts p                       #=> Point(@x=1, @y=0)
+```
+
+Notice that the chained calls of `move_right` returned the expected result, but only the first call to it modified the point `p`, as the second call was operating on the _copy_ of the struct that was passed to it from the first call, and this copy was discarded after the result of the expression was printed.
```

Comment:
Present tense reads better:
```suggestion
Notice that the chained calls of `move_right` return the expected result, but only the first call to it modifies the variable `p`, as the second call operates on the _copy_ of the struct passed to it from the first call, and this copy is discarded after printing the result of the expression.
```

Reply:
"and this copy is discarded after printing the result of the expression."
When we remove `puts` from the statement 
`puts p.move_right.move_right #=> Point(@x=2, @y=0)`
above, the ending of the sentence is not correct as it is not clear what 'printing' it talks about.

Reply:
"discarded after the expression is executed" ?

Reply:
Changed in a new commit.